File: app/src/main/java/com/innovatube/boilerplate/ui/base/BaseActivity.java
package com.innovatube.boilerplate.ui.base;

import android.support.v7.app.AppCompatActivity;

import com.innovatube.boilerplate.InnovatubeApplication;
import com.innovatube.boilerplate.injection.component.ActivityComponent;
import com.innovatube.boilerplate.injection.component.DaggerActivityComponent;


/**
 * Created by TOIDV on 4/4/2016.
 */
public class BaseActivity extends AppCompatActivity {

    ActivityComponent mActivityComponent;

    public ActivityComponent getActivityComponent() {
        if (mActivityComponent == null) {
            mActivityComponent = DaggerActivityComponent.builder()
                    .applicationComponent(InnovatubeApplication.get(this).getComponent())
                    .build();
        }
        return mActivityComponent;
    }
}

File: lib/android/boilerplate/templates/boilerplate/app/src/main/java/com/innovatube/boilerplate/presentation/base/BaseActivity.kt
package <%= package_name %>.presentation.base

import android.support.v7.app.AppCompatActivity
import android.support.v7.app.AppCompatDelegate
import <%= package_name %>.InnovatubeApplication
import <%= package_name %>.util.di.components.ActivityComponent
import <%= package_name %>.util.di.modules.ActivityModule


abstract class BaseActivity : AppCompatActivity() {

    val component: ActivityComponent by lazy {
        (application as InnovatubeApplication).getComponent().plus(ActivityModule(this))
    }

    companion object {
        init {
            AppCompatDelegate.setCompatVectorFromResourcesEnabled(true)
        }
    }
}
